apps/verticalhome/test/marionette/app_launch_visibility_test.js

```diff
@@ -0,0 +1,75 @@
+/* global __dirname */
+'use strict';
+
+var assert = require('assert');
+var AppInstall =
+  require('../../../../apps/system/test/marionette/lib/app_install');
+var iconAppState = require('./lib/icon_app_state');
+var createAppServer = require('./server/parent');
+
+marionette('Vertical Home - Launch visibility', function() {
+  var client = marionette.client(require(__dirname + '/client_options.js'));
+  var server;
+  setup(function(done) {
+    var app = __dirname + '/fixtures/template_app';
+    createAppServer(app, client, function(err, _server) {
+      server = _server;
+      done(err);
+    });
+  });
+
+  var subject;
+  var system;
+  var appInstall;
+
+  setup(function() {
+    subject = client.loader.getAppClass('verticalhome');
+    system = client.loader.getAppClass('system');
+    appInstall = new AppInstall(client);
+
+    system.waitForStartup();
+    subject.waitForLaunch();
+
+    client.switchToFrame();
+    appInstall.installPackage(server.packageManifestURL);
+
+    client.switchToFrame(system.getHomescreenIframe());
+  });
+
+  teardown(function(done) {
+    server.close(done);
+  });
+
+  test('Update overflow for fast resume once hidden', function() {
```

Comment:
My objection to this test is that it doesn't actually test that overflow is hidden during the animation, only that it is hidden when an application is launched and removed when the home button is pressed. I think such a test would probably be easier as a unit test, and no less valuable... As a sanity-check, I guess this works, but I question its usefulness.